gui/system/models.py

```diff
@@ -1212,5 +1238,82 @@ def is_enabled(self):
         return self.is_available(support=self)[0] and self.enabled
 
 
+class ACMERegistrationBody(Model):
+    contact = models.EmailField(
+        verbose_name=_('Contact Email')
+    )
+    status = models.CharField(
+        verbose_name=_('Status'),
+        max_length=10
+    )
+    key = models.TextField(
+        verbose_name=_('JWKRSAKey')
+    )
+    acme = models.ForeignKey(
+        'ACMERegistration',
+        on_delete=models.CASCADE,
+    )
+
+
+class ACMERegistration(Model):
+    uri = models.URLField(
+        verbose_name=_('URI')
+    )
+    directory = models.URLField(
+        verbose_name=_('Directory URI'),
+        unique=True
+    )
+    tos = models.URLField(
+        verbose_name=_('Terms of Service')
+    )
+    new_account_uri = models.URLField(
+        verbose_name=_('New Account Uri')
+    )
+    new_nonce_uri = models.URLField(
+        verbose_name=_('New Nonce Uri')
+    )
+    new_order_uri = models.URLField(
+        verbose_name=_('New Order Uri')
+    )
+    revoke_cert_uri = models.URLField(
+        verbose_name=_('Revoke Certificate Uri')
+    )
+
+
+class DNSAuthenticator(Model):
```

Comment:
It just came to me that DNSAuthenticator may be a bad name for it.

If I just see the label "DNS Authenticator" in the UI the user will go: "DNS Authenticator? Authenticator of what? What does it do?"

Wont you agree? What if we called it "ACME DNS Authenticator" ?